packages/codelabs/src/basics/lit-html.md

```diff
@@ -0,0 +1,1139 @@
+          `;
+        }
+
+        render() {
+          if (!this.todos) {
+            return html``;
+          }
+
+          return html`
+            <ol>
+              ${this.todos.map(
+                todo => html`
+                  <li>
+                    <input
+                      type="checkbox"
+                      .checked=${todo.finished}
+                      @change=${e => this._changeTodoFinished(e, todo)}
+                    />
+                    ${todo.text}
+                    <button @click=${() => this._removeTodo(todo)}>X</button>
+                  </li>
+                `,
+              )}
+            </ol>
+          `;
+        }
+
+        _changeTodoFinished(e, changedTodo) {
+          const eventDetails = { changedTodo, finished: e.target.checked };
+          this.dispatchEvent(new CustomEvent('change-todo-finished', { detail: eventDetails }));
+        }
+
+        _removeTodo(todo) {
+          this.dispatchEvent(new CustomEvent('remove-todo', { detail: todo }));
+        }
+      }
+
+      customElements.define('todo-list', TodoList);
+
+      class TodoApp extends LitElement {
+        static get properties() {
+          return {
+            todos: { type: Array },
+          };
+        }
+
+        constructor() {
+          super();
+
+          this.todos = [
+            { text: 'Do A', finished: true },
+            { text: 'Do B', finished: false },
+            { text: 'Do C', finished: false },
+          ];
+        }
+
+        render() {
+          const finishedCount = this.todos.filter(t => t.finished).length;
+          const unfinishedCount = this.todos.length - finishedCount;
+
+          return html`
+            <h1>Todo list</h1>
+
+            <input id="addTodoInput" placeholder="Name" /><button @click=${this._addTodo}>
+              Add todo
+            </button>
+
+            <todo-list
+              .todos=${this.todos}
+              @change-todo-finished=${this._changeTodoFinished}
+              @remove-todo=${this._removeTodo}
+            ></todo-list>
+
+            <p>Total finished: ${finishedCount}</p>
+            <p>Total unfinished: ${unfinishedCount}</p>
+          `;
+        }
+
+        _changeTodoFinished(e) {
+          const { changedTodo, finished } = e.detail;
+
+          this.todos = this.todos.map(todo => {
+            if (todo !== changedTodo) {
+              return todo;
+            }
+            return { ...changedTodo, finished };
+          });
+        }
+
+        _addTodo() {
+          const input = this.shadowRoot.getElementById('addTodoInput');
+          const text = input.value;
+          input.value = '';
+
+          this.todos = [...this.todos, { text, finished: false }];
+        }
+
+        _removeTodo(e) {
+          this.todos = this.todos.filter(todo => todo !== e.detail);
+        }
+      }
+
+      customElements.define('todo-app', TodoApp);
+    </script>
+  </body>
+</html>
+```
+
+</detail>
+
```

Comment:
Maybe we can 'wrap up' by summarizing what we did/learned in this codelab, and maybe where to go next? The end feels a little 'abrupt' to me now.